### PR TITLE
Exposing addIdentity method from hmat

### PR DESCRIPTION
**Pull request by poncetr**
Hi,
Please, review and pull if everything ok.
I need this to correctly use HMatrix for kriging applications.

### CODE REVIEW

File: lib/src/Base/Stat/HMatrixImplementation.cxx [hunk 1]
```diff
@@ -253,6 +253,15 @@ HMatrixImplementation::assemble(const HMatrixRealAssemblyFunction& f, char symme
 #endif
 }
 
+void HMatrixImplementation::addIdentity(Scalar alpha)
+{
+#ifdef OPENTURNS_HAVE_HMAT
+  static_cast<hmat_interface_t *>(hmatInterface_)->add_identity(static_cast<hmat_matrix_t *>(hmat_), &alpha);
+#else
+  throw NotYetImplementedException(HERE) << "OpenTURNS had been compiled without HMat support";
```
Comment (jschueller) [inline lib/src/Base/Stat/HMatrixImplementation.cxx]: has been, please fix other occurences

File: lib/src/Base/Stat/openturns/HMatrix.hxx [hunk 2]
```diff
@@ -80,6 +80,8 @@ class OT_API HMatrix :
   /** Get the diagonal */
   Point getDiagonal() const;
 
+  void addIdentity(Scalar alpha);
```
Comment (jschueller) [inline lib/src/Base/Stat/openturns/HMatrix.hxx]: you can add a small comment

**Comment by regislebrun** [comment]
+1

**Comment by poncetr** [comment]
Thanks. You can pull now.

**Pull request merged**